### PR TITLE
Closes #5022 - Add BrowserMenuHighlightableSwitch

### LINKED ISSUE #5022: Allow highlightable / toggleable toolbar items
As far as I can tell, the toolbar API currently supports toggleable items via `BrowserMenuImageSwitch`, and highlightable items via `BrowserMenuHighlightableItem`, but no provides no way to create a toggleable, highlightable item.  This is required for [Fenix #4281](https://github.com/mozilla-mobile/fenix/issues/4281) ([also see this relevant mock](
https://app.goabstract.com/share/7cd59d53-ad3a-4b33-8131-74703c1859a2?collectionLayerId=919946d7-808d-4239-a7e9-b49493929b31&mode=design&present=true&preview=false&sha=eff2146ca3977be955d41b7b04dc4c8e799d476d)).

In the long term, this points to an oncoming combinatorial explosion of toolbar options, but I'm not sure how best to solve that problem.

### PR DESCRIPTION
**Pull request by NotWoods**
New component for Reader Mode in Fenix.

---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by NotWoods** [comment]
bors r=Baron-Severin

**Comment by NotWoods** [comment]
I'm planning to make bigger menu changes down the line so I think this is fine for now.

File: components/browser/menu/src/main/java/mozilla/components/browser/menu/BrowserMenuHighlight.kt [hunk 1]
```diff
@@ -59,3 +59,13 @@ sealed class BrowserMenuHighlight {
         override val label: String? = null
     }
 }
+
+/**
+ * Indicates that a menu item shows a highlight.
+ */
+interface HighlightableMenuItem {
+    /** Highlight object representing how the menu item will be displayed when highlighted. */
```
Comment (jonalmeida) [inline components/browser/menu/src/main/java/mozilla/components/browser/menu/BrowserMenuHighlight.kt]: nit: docs formatting.

File: components/browser/menu/src/main/res/layout/mozac_browser_menu_highlightable_switch.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        android:translationY="@dimen/mozac_browser_menu_highlightable_notification_translate_y"
+        android:background="@android:color/transparent"
+        android:importantForAccessibility="no"
+        android:visibility="gone"
+        app:layout_constraintTop_toTopOf="@id/image"
+        app:layout_constraintEnd_toEndOf="@id/image"
+        app:srcCompat="@drawable/mozac_browser_menu_notification_icon"
+        tools:visibility="visible" />
+
+    <Switch
```
Comment (jonalmeida) [inline components/browser/menu/src/main/res/layout/mozac_browser_menu_highlightable_switch.xml]: SwitchCompat?

**Comment by NotWoods** [comment]
@jonalmeida We plan to make bigger changes down the line (#5373). This is just to support Reader Mode until we have a chance to make bigger menu changes down the line.

**Comment by NotWoods** [comment]
bors r=Amejia481,Baron-Severin

**Comment by NotWoods** [comment]
bors r=Amejia481,Baron-Severin

**Comment by NotWoods** [comment]
bors retry

**Pull request merged**